kernel/src/hil/touch.rs

```diff
@@ -0,0 +1,83 @@
+//! Interface for touch input devices
+
+use crate::ReturnCode;
+
+#[derive(Debug, Copy, Clone)]
+pub enum TouchStatus {
+    Pressed,
+    Released,
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum GestureEvent {
+    MoveUp,
+    MoveDown,
+    MoveLeft,
+    MoveRight,
+    ZoomIn,
+    ZoomOut,
+}
+
+#[derive(Copy, Clone)]
+pub struct TouchEvent {
+    pub status: TouchStatus,
+    // touch (x, y) position
+    pub x: usize,
+    pub y: usize,
+
+    // touch id, value defined by the driver
+    pub id: usize,
+
+    // touch area
+    pub area: Option<usize>,
+
+    // touch weight
+    pub weight: Option<usize>,
+}
+
+pub trait Touch {
+    // enable touch
+    fn enable(&self) -> ReturnCode;
+
+    // disable touch
+    fn disable(&self) -> ReturnCode;
```

Comment:
Is the purpose of enable/disable to start/stop events?

Reply:
It enables and disables the touch panel, this is useful for low power